### PR TITLE
ref(control_silo): Consolidate User models (and authenticator) and tests

### PR DESCRIPTION
**Pull request by Christinarlong**
Moves all User and authenticator specific models and tests to their own directories.
    
ref(#73856)

### CODE REVIEW

File: bin/mock-user [hunk 1]
```diff
@@ -8,7 +8,7 @@ import argparse
 
 
 def main(username, newsletter_consent_prompt=None):
-    from sentry.models.user import User
+    from sentry.users.models.users.user import User
```
Comment (markstory) [inline bin/mock-user:11]: Could this be `sentry.users.models.user`? Or is the additional layer of modules helpful?

File: src/sentry/models/__init__.py [hunk 2]
```diff
@@ -1,3 +1,7 @@
+from ..users.models.auth.authenticator import *  # NOQA
+from ..users.models.users.userip import *  # NOQA
+from ..users.models.users.userreport import *  # NOQA
+from ..users.models.users.userrole import *  # NOQA
```
Comment (markstory) [inline src/sentry/models/__init__.py:4]: I would avoid relative imports that go up the module path. These could be `from sentry.users.models ...`.

File: src/sentry/models/authenticator.py [hunk 3]
```diff
@@ -1,204 +1,3 @@
-from __future__ import annotations
+from ..users.models.auth.authenticator import Authenticator  # NOQA
```
Comment (markstory) [inline src/sentry/models/authenticator.py:1]: Could we do `from sentry.users.models.authenticator import Authenticator`. The users app will have fewer than 10 models in it and I don't think the additional layer of directories adds much.

File: src/sentry/models/authenticator.py [hunk 4]
```diff
@@ -1,204 +1,3 @@
...
-
-    @classmethod
-    def sanitize_relocation_json(
-        cls, json: Any, sanitizer: Sanitizer, model_name: NormalizedModelName | None = None
-    ) -> None:
-        model_name = get_model_name(cls) if model_name is None else model_name
-        super().sanitize_relocation_json(json, sanitizer, model_name)
-
-        sanitizer.set_string(json, SanitizableField(model_name, "config"), lambda _: '""')
+# flake8 dislikes imported but unused modules
```
Comment (markstory) [inline src/sentry/models/authenticator.py:3]:
```suggestion
__all__ = ("Authentictator", )
```

Giving the module an `__all__` is what I've done in the past for these. It also lets you skip the `# NOQA` on the import.

File: src/sentry/models/avatars/__init__.py [hunk 5]
```diff
@@ -1,9 +1,9 @@
+from ...users.models.users.user_avatar import UserAvatar
```
Comment (markstory) [inline src/sentry/models/avatars/__init__.py:1]: Could we use absolute imports when we're crossing modules that are children of `sentry`?

File: src/sentry/models/userrole.py [hunk 6]
```diff
@@ -1,97 +1,7 @@
...
+from sentry.users.models.users.userrole import (  # NOQA
+    UserRole,
+    UserRoleUser,
     manage_default_super_admin_role,
-    dispatch_uid="manage_default_super_admin_role",
-    weak=False,
-    sender=SiloMode.MONOLITH,
 )
+
+# flake8 dislikes imported but unused modules
```
Comment (markstory) [inline src/sentry/models/userrole.py:7]: This is another place where I would put an `__all__` to appease tooling and humans doing `*` imports.

File: src/sentry/users/models/__init__.py [hunk 7]
```diff
@@ -0,0 +1,6 @@
+from ...users.models.users.user import *  # NOQA
```
Comment (markstory) [inline src/sentry/users/models/__init__.py:1]: Could we use absolute imports here? The `...` pattern is hard to search for in the future.

File: bin/mock-user (hunk 1 above)
Comment (Christinarlong) [reply bin/mock-user:11]: Sure it definitely could go there. Just kinda worried that with all the other models, the models directory could grow p. large though we could do something like:
```
sentry.users.models
|
|--> auth
         |
         |authenticator.py
         
|--> emails
        
| user.py
| useremail.py
```
does this feel cleaner/makes more sense?

Comment (Christinarlong) [reply bin/mock-user:11]: nvm, read the rest of the review. 10 is p. few so I'm down to just flatten and put everything in models

**Comment by Christinarlong** [comment]
Breaking down into smaller PRs

**Pull request closed without merging**